backend/src/main/java/de/neuefische/app/spotify/SpotifyApiService.java

```diff
@@ -40,6 +49,50 @@ public void createNewSpotifyPlaylist(String title, List<String> uris, String spo
         addTracksToNewPlaylist(accessTokenResponse.getBody(), newPlaylistId, uris);
     }
 
+    public Optional<PlaylistData> getPlaylistFromSpotify(String id){
+        ResponseEntity<SpotifyGetAccessTokenBody> accessTokenResponse = getRefreshTokenFromSpotify();
+
+        try {
+            ResponseEntity<SpotifyGetPlaylistBody> userPlaylistsTracksResponse = restTemplate.exchange(
+                    "https://api.spotify.com/v1/playlists/" + id,
+                    HttpMethod.GET,
+                    new HttpEntity<>(createHeaders(accessTokenResponse.getBody().accessToken())),
+                    SpotifyGetPlaylistBody.class
+            );
+            SpotifyGetPlaylistBody responseBody = userPlaylistsTracksResponse.getBody();
+            List<PlaylistTrack> tracks = responseBody.tracks().items().stream().map(item -> PlaylistTrack.of(item.track())).toList();
+            List<PlaylistImage> images = responseBody.images().stream().map(image -> PlaylistImage.of(image)).toList();
+            String urlForNextTracks = responseBody.tracks().next();
+            boolean hasMoreThan100Tracks = !Objects.equals(urlForNextTracks, null) && responseBody.tracks().total() > 100;
+            while(hasMoreThan100Tracks){
+                ResponseEntity<SpotifyPlaylistTracks> userPlaylistsNextTracksResponse = restTemplate.exchange(
```

Comment:
The code inside the `while`-loop is the same as before the loop. Think about moving it to a private method, that you can call before and inside the loop.

Reply:
As I can see only the restTemplate.exchange is repeated - but actually the requests are directed to two different endpoints of the Spotify Api - so the url is different and the ResponseEntity classes that I get are different.